sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CallToAsyncMethodShouldNotBeBlocking.cs

```diff
@@ -92,11 +93,30 @@ private Task SleepTask()
             Thread.Sleep(10);
         }
 
+        public void Run(Task<int> task)
+        {
+            Action<Task<int>> arg = (action) =>
+            {
+                var ret = action.Result; // Noncompliant FP, we do not track actions which are used on ContinueWith
+            };
+
+            task.ContinueWith(arg);
+        }
+
+        // See https://github.com/SonarSource/sonar-dotnet/issues/2413
         public Task<string> Run(Task<string> task)
         {
             return task.ContinueWith(completedTask =>
             {
-                return completedTask.Result; // Noncompliant FP, task is already completed at this point.
+                return completedTask.Result; // Compliant, task is already completed at this point.
```

Comment:
IsContinueWithCallWithArgumentName is great. Please add test to demonstrate it 🙂 
```C#
Task<int> anotherTask = null; // Pretend to compute something
var b = anotherTask.Result; // Noncompliant, this task is not safe inside ContinueWith
```